crates/matrix-sdk/src/event_cache/room/mod.rs

```diff
@@ -587,13 +591,75 @@ mod private {
             Ok(Self { room, store, events, waited_for_initial_prev_token: false })
         }
 
+        /// Removes the bundled relations from an event, if they were present.
+        ///
+        /// Only replaces the present if it contained bundled relations.
+        ///
+        /// Returns an `Option` so as to be able to short-circuit with `?`, but
+        /// the result is meaningless and mustn't be used.
+        fn maybe_strip_relations<T>(event: &mut Raw<T>) -> Option<()> {
+            // We're going to get rid of the `unsigned`/`m.relations` field, if it's
+            // present.
+            let mut val: serde_json::Value = event.deserialize_as().ok()?;
+            let unsigned = val.get_mut("unsigned")?;
+            let unsigned_obj = unsigned.as_object_mut()?;
+            if unsigned_obj.remove("m.relations").is_some() {
+                *event = Raw::new(&val).ok()?.cast();
+            }
+            None
+        }
+
+        /// Strips the bundled relations from a collection of events.
+        fn strip_relations_from_events(items: &mut [SyncTimelineEvent]) {
+            for ev in items.iter_mut() {
+                match &mut ev.kind {
+                    TimelineEventKind::Decrypted(decrypted) => {
+                        // Remove all information about encryption info for
+                        // the bundled events.
+                        decrypted.unsigned_encryption_info = None;
+
+                        // Remove the `unsigned`/`m.relations` field, if needs be.
+                        let _ = Self::maybe_strip_relations(&mut decrypted.event);
+                    }
+
+                    TimelineEventKind::UnableToDecrypt { event, .. }
+                    | TimelineEventKind::PlainText { event } => {
+                        let _ = Self::maybe_strip_relations(event);
+                    }
+                }
+            }
+        }
+
         /// Propagate changes to the underlying storage.
         async fn propagate_changes(&mut self) -> Result<(), EventCacheError> {
             let updates = self.events.updates().take();
 
             if !updates.is_empty() {
                 if let Some(store) = self.store.get() {
                     let locked = store.lock().await?;
+
+                    // Strip relations from the `PushItems` updates.
+                    let updates = updates
+                        .into_iter()
```

Comment:
Curious: did you try this with `iter_mut` to modify this in-place? Or do you prefer this pattern?

Reply:
Good catch, thanks!